books/free-programming-books-subjects.md

```diff
@@ -99,6 +99,7 @@ Books that cover a specific programming language can be found in the  [BY PROGRA
 * [Data Structures and Algorithms: Annotated Reference with Examples](https://web.archive.org/web/20170715160229/http://dotnetslackers.com/Community/files/folders/data-structures-and-algorithms/entry30283.aspx) - G. Barnett and L. Del Tongo
 * [Data Structures Succinctly Part 1, Syncfusion](https://www.syncfusion.com/resources/techportal/ebooks/datastructurespart1) (PDF, Kindle) (email address *requested*, not required)
 * [Data Structures Succinctly Part 2, Syncfusion](https://www.syncfusion.com/resources/techportal/ebooks/datastructurespart2) (PDF, Kindle) (email address *requested*, not required)
+* [Dynamic Programming for Coding Interviews: A Bottom-Up Approach to Problem](https://libgen.is/book/index.php?md5=CCAA5C6429A0954D69371D5B2B411060) (PDF) 
```

Comment:
- Domain not allowed. Use author/publisher website
- Not free 🤔?? https://www.amazon.com/-/en/Meenakshi/dp/1946556696
- Missing author attributions